tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -206,5 +208,26 @@ private static void WriteRequestHeaders(WebRequest request, IDictionary<string,
             var proxyInstance = obj.DuckAs<ILambdaContext>();
             return proxyInstance?.ClientContext?.Custom;
         }
+
+        internal static void InjectTraceContext(object executionContext, string traceId, string parentId, string samplingPriority)
+        {
+            var clientContextString = $"{{ \"custom\": {{ \"x-datadog-trace-id\": \"{traceId}\", \"x-datadog-parent-id\": \"{parentId}\", \"x-datadog-sampling-priority\": \"{samplingPriority}\" }}}}";
+
+            var proxyInstance = executionContext.DuckAs<IExecutionContext>();
+            proxyInstance.RequestContext.OriginalRequest.ClientContext = clientContextString;
+
+            var plainTextBytes = System.Text.Encoding.UTF8.GetBytes(clientContextString);
+            var clientContextB64 = System.Convert.ToBase64String(plainTextBytes);
+            proxyInstance.RequestContext.OriginalRequest.ClientContextBase64 = clientContextB64;
+        }
+
+        internal static bool IsSyncLambdaInvocation(object executionContext)
```

Comment:
suggestion: To make this more efficient, you can use the same proxy objects from the instrumentation methods instead of recreating the `proxyInstance` in this method

```suggestion
        internal static bool IsSyncLambdaInvocation<TExecutionContext>(TExecutionContext executionContext)
            where TExecutionContext : IExecutionContext
```

Reply:
implemented